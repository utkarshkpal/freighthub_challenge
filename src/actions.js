import axios from "axios";
axios.defaults.headers.put["Content-Type"] = "application/json";

export function updateNameAction(newData) {
  return new Promise((resolve, reject) => {
    const url = `http://localhost:3004/shipments/${newData.id}`;
    axios
      .put(url, newData)
      .then(resp => {
        resolve(resp.data, newData.id);
      })
      .catch(error => {
        reject(error);
      });
  });
}

export function getShipmentDataById(id) {
  return new Promise((resolve, reject) => {
    const url = `http://localhost:3004/shipments/${id}`;
    axios
      .get(url)
      .then(resp => {
        console.log(resp.data);
        resolve(resp.data);
      })
      .catch(error => {
        reject(error);
      });
  });
}
