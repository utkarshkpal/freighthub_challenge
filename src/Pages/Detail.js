import React, {Component} from 'react';
import { PageHeader, Tag, Tabs, Button, Statistic, Row, Col } from 'antd';
import Cargo from '../Component/Cargo';
import {getShipmentDataById} from '../actions';
import { identifier } from '@babel/types';



const Content = ({origin,destination}) =>{
   
    return(
    
    <div className='elementBorder'>
    <Statistic title="Origin" value={origin}/>
    <Statistic title="Destination" value={destination}/>
    </div>
  )

}
const ExtraContent = ({total,userId}) =>{
   
    return(
        <Row>
        <Col span={12}>
          <Statistic className='elementBorder' title="UserId" value={userId} />
        </Col>
        <Col span={12}>
          <Statistic className='elementBorder' title="Total" value={total} />
        </Col>
      </Row>)

}


  
class Detail extends Component {

    state = {
        shipment:{}
    }
    componentDidMount() {

      if(this.props.match){
        const {match:{params:{id}}}= this.props;
        getShipmentDataById(id).then((res)=>{
           this.setState({shipment:res});
        });
      }
        
    }
    

    renderTag = () => {
        const {status} = this.state.shipment;
        let color = 'yellow';
        if(status.toUpperCase() == 'COMPLETED') {
            color = 'green'
        } 
        return <Tag color={color}>{status}</Tag>;
    }

    render() {
        const {shipment} = this.state;
        const { history} = this.props;
                if(Object.keys(shipment).length){
            const {
                id,name,cargo,mode,type,destination,origin,services,total,userId
            } = shipment;
            console.log(this.props.match.params.id);
            return (
              <>
               <div className='headerWrapper elementBorder'>
               <h2>Shipment Detail</h2>
               </div>
               <div className='detailContent'>
                <PageHeader
                className='detailHeader'
                onBack={() => history.push('/')}
                title={id+' ' + name}
                tags={this.renderTag()}
                extra={[
                  <span>Type : </span> ,
                  <Button key="2">{type}</Button>,
                  <span>Mode : </span> ,
                  <Button key="1" type="primary">
                    {mode.toUpperCase()}
                  </Button>
                ]}
              > 
                <Row>
                <Col span={1}/>
                    <Col span={10}>
                        <div className="wrap">
                        <div className="content padding"><Content origin={origin} destination={destination}/></div>
                        <div className="extraContent">{<ExtraContent total={total} userId={userId}/>}</div>
                        </div>
                    </Col>
                    <Col span={1} />
                    <Col span={12}>
                        <Cargo cargo={cargo}/>
                    </Col>
                </Row>
              </PageHeader>
              </div>
              </>
            );

        }else{
            return null;
        }
        
    }
}

export default Detail;