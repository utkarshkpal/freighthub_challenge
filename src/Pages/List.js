import React, { Component } from "react";
import Table from "../Component/Table";
import { updateExpression } from "@babel/types";
import { updateNameAction } from "../actions";
import { message } from "antd";
import { PageHeader } from "antd";

class List extends Component {
  state = {
    shipments: []
  };

  updateName = newData => {
    updateNameAction(newData)
      .then(newShipmentData => {
        message.success("Changes Saved");
      })
      .catch(error => {
        console.log(error);
      });
  };

  render() {
    const { history} = this.props;
    const { shipments } = this.state;
    return (
      <>
        <div className='headerWrapper elementBorder'>
               <h2>Shipment List</h2>
        </div>
        {shipments ? (
          <div className='tableWrapper'>
          <Table history={history} updateName={this.updateName} shipments={shipments} />
          </div>
        ) : null}
      </>
    );
  }
}

export default List;
