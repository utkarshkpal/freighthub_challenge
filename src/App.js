import React from 'react';
import logo from './logo.svg';
import Detail from './Pages/Detail';
import List from './Pages/List';
import { BrowserRouter, Route, Link,Switch } from "react-router-dom";


function App() {

return (
<div className="App">
  <div className='content'>
  <Switch>
    <Route exact path="/" component={List} />
    <Route path="/detail/:id" component={Detail} />
  </Switch>
  </div>
   </div>
  );
}

export default App;
