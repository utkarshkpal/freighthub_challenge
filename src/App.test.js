import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { mount,shallow,render } from 'enzyme';
import Detail from './Pages/Detail';
import List from './Pages/List';
import Link from 'react-router-dom';
import {getShipmentDataById} from './actions.js';



describe('App', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<App debug />);
  
    expect(component).toMatchSnapshot();
  });
});

describe('Detail', () => {
  it('should render properly', () => {
    const component = shallow(<Detail/>);
  
    expect(component).toMatchSnapshot();
  });
});

describe('List', () => {
  it(' should render properly', () => {
    const component = shallow(<List/>);
  
    expect(component).toMatchSnapshot();
  });
});

// describe('List', () => {
//   it('button click should hide component', () => {
//     const component = mount(<List/>);
//     component
//       .find('.sid')
//       .simulate('click');
//     expect(getShipmentDataById).toHaveBeenCalled();
//     component.unmount();
//   });
 
// });

