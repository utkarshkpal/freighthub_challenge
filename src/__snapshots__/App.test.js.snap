// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;

exports[`Detail should render properly 1`] = `ShallowWrapper {}`;

exports[`List  should render properly 1`] = `ShallowWrapper {}`;

exports[`MyComponent should render correctly in "debug" mode 1`] = `ShallowWrapper {}`;
